### ISSUE TITLE
Allow to specificate the tag name at git flow (hotfix|release) finish

### ISSUE THREAD

**Issue by derpue** [petervanderdoes/gitflow-avh#199]
Currently it uses the name given at `git flow (hotfix|release) start` as tag name during the finish stage, thus the branch name and the tag have to be the same name.

But we like to have speaking names for the branches, but only a short version for the tag. So we would like to have something like:
`git flow hotfix start SomeNastyBug` to get branch hotfix/SomeNastyBug
and later finish it with something like
`git flow hotfix finish SomeNastyBug 2.4.1` to get the merge result taged as v2.4.1 in master

**Comment by jawshooah** [comment]
:+1: Hotfix branches should be named based on what they're fixing.

**Comment by ByScripts** [comment]
:+1:

But it would be great if the branch name argument was still optional (`git flow hotfix finish` without argument uses the current branch name).

Maybe using a dot? (but visually, it's not really obvious)

`git flow hotfix finish . 2.4.1`

Or a double-dash?

`git flow hotfix finish -- 2.4.1`

Or something else...

On the other hand, since branches are deleted once merged, and a hotfix is meant to be developed "quick and dirty"... maybe it's not really important to name the branch "2.4.1".

If the fix is complex and will be worked on for days... maybe we should use a "bugfix" branch, not a "hotfix" one

**Comment by JorisInsign** [comment]
:+1: 
We would like to have this option too !

**Comment by tzewangdorje** [comment]
This caught us out too at first. We would also like to use a friendly feature branch name during development and then tag v1.4.3 etc when we finish the hotfix.

**Comment by jrial** [comment]
Ditto. In our workflow, branch names are descriptive (refer to issue # + short description), while tags represent the release version numbers.

**Comment by jrial** [comment]
@petervanderdoes this is itching me, and I'm ready to scratch it. What syntax do you prefer?

`git flow hotfix finish branchname versionnumber`
or
`git flow hotfix finish branchname -v versionnumber`

**Comment by petervanderdoes** [comment]
I'm currently on transistion from one city to another,  but wanted to quickly add my view. We could add something like

`git flow hotfix finish --tag <tagname> <branchname>`

If that flag is used the normal tagging would be overridden by the value of the tag.

In the end it means we would change the behavior of several commands, instead of using <version> it would become <branch> which I'm totally for but it does mean several other changes beside just this one.

I probably will be back in full force around the first week of July.

**Comment by jrial** [comment]
As far as I see, the only impact is during the following, correct?

- hotfix finish
- release finish

The feature and bugfix subcommands don't concern themselves with tagging ATM. Nor should they, so no changes required there.

**Comment by rendekarf** [comment]
Hey guys, is the pr going to be accepted ? Would be really nice to state a different tag name as compared to the release/hotfix branch names

**Comment by petervanderdoes** [comment]
Implemented per [Release 1.10.0](https://github.com/petervanderdoes/gitflow-avh/releases/tag/1.10.0)

**Comment by aesyondu** [comment]
Save you a click:
```
Add new flag -T, --tagname to override the tag name when finishing a release or hotfix.
```
